packages/core/src/scores/scoreBatchTraces/scoreBatchTracesWorkflow.ts

```diff
@@ -0,0 +1,233 @@
+  scorer: MastraScorer;
+  target: { traceId: string; spanId?: string };
+  tracingContext: TracingContext;
+}) {
+  // TODO: add storage api to get a single span
+  const trace = await storage.getAITrace(target.traceId);
+
+  if (!trace) {
+    throw new Error(`Trace not found for scoring, traceId: ${target.traceId}`);
+  }
+
+  let span: AISpanRecord | undefined;
+  if (target.spanId) {
+    span = trace.spans.find(span => span.spanId === target.spanId);
+  } else {
+    span = trace.spans.find(span => span.parentSpanId === null);
+  }
+
+  if (!span) {
+    throw new Error(
+      `Span not found for scoring, traceId: ${target.traceId}, spanId: ${target.spanId ?? 'Not provided'}`,
+    );
+  }
+
+  const scorerRun = buildScorerRun({
+    scorerType: scorer.type === 'agent' ? 'agent' : undefined,
+    tracingContext,
+    trace,
+    targetSpan: span,
+  });
+
+  const result = await scorer.run(scorerRun);
+  const traceId = `${target.traceId}${target.spanId ? `-${target.spanId}` : ''}`;
+  const scorerResult = {
+    ...result,
+    scorer: {
+      id: scorer.name,
+      name: scorer.name,
+      description: scorer.description,
+    },
+    traceId,
+    entityId: span.name,
+    entityType: span.spanType,
+    entity: { traceId: span.traceId, spanId: span.spanId },
+    source: 'TEST',
+    scorerId: scorer.name,
+  };
+
+  const savedScoreRecord = await validateAndSaveScore({ storage, scorerResult });
+  await attachScoreToSpan({ storage, span, scoreRecord: savedScoreRecord });
+}
+
+async function validateAndSaveScore({ storage, scorerResult }: { storage: MastraStorage; scorerResult: ScorerRun }) {
+  const payloadToSave = saveScorePayloadSchema.parse(scorerResult);
+  const result = await storage.saveScore(payloadToSave);
+  return result.score;
+}
+
+function buildScorerRun({
+  scorerType,
+  tracingContext,
+  trace,
+  targetSpan,
+}: {
+  scorerType?: string;
+  tracingContext: TracingContext;
+  trace: AITraceRecord;
+  targetSpan: AISpanRecord;
+}) {
+  let runPayload: ScorerRun;
+  if (scorerType === 'agent') {
+    runPayload = {
+      input: transformTraceToScorerInput(trace as any),
+      output: transformTraceToScorerOutput(trace as any),
+    };
+  } else {
+    runPayload = { input: targetSpan.input, output: targetSpan.output };
+  }
+
+  runPayload.tracingContext = tracingContext;
+  return runPayload;
+}
+
+async function attachScoreToSpan({
+  storage,
+  span,
+  scoreRecord,
+}: {
+  storage: MastraStorage;
+  span: AISpanRecord;
+  scoreRecord: ScoreRowData;
+}) {
+  const existingLinks = span.links || [];
+  const link = {
+    type: 'score',
+    scoreId: scoreRecord.id,
+    scorerName: scoreRecord.scorer.name,
+    score: scoreRecord.score,
+    createdAt: scoreRecord.createdAt,
+  };
+  await storage.updateAISpan({
+    spanId: span.spanId,
+    traceId: span.traceId,
+    updates: { links: [...existingLinks, link] },
+  });
+}
+
+export const scoreBatchTracesWorkflow = createWorkflow({
```

Comment:
do we need 'batch' in the exported name?  doesn't plural 'Traces' infer it's a batch?